### ISSUE TITLE
Issue running single-host-cpu profile on docker Windows

### ISSUE THREAD

**Issue by jericko-ag** [adalkiran/distributed-inference#1]
Currently getting this error in inference:
I checked the container and it uses 3.12.x for libprotoc (which should be fine as shown in the message?)
![image](https://github.com/user-attachments/assets/08f4d33c-f61f-46b1-a584-1c03613a4dbc)
Updating the container to 3.20 via:
`
pip install protobuf==3.20
`

just opened a whole new can of worms:

![image](https://github.com/user-attachments/assets/5bd5b239-152e-475f-b4d6-c59dbb82eb54)

Did a dirty hack of downgrading the container again:
`
pip install numpy==1.23.4
`

This time it shows:

![image](https://github.com/user-attachments/assets/0b122007-e0b0-4797-b504-b7dc594406db)

On the UI, it start after clicking 'create peer connection', but then disconnects after 15 second-ish:

![image](https://github.com/user-attachments/assets/3272bbd7-12fd-4312-b505-51f5b40a5464)

.env remains default except for this change:
`
![image](https://github.com/user-attachments/assets/d9a347e4-5732-44ef-88f0-0928576c80e1)

`

All these issues aside, have you by any chance, implemented an inference that uses RT-DETR?

**Comment by adalkiran** [comment]
Hi, thanks for your detailed report. I remember that I've intentionally downgraded one or more library versions to make it work during development. Some changes might be made on dependencies since a long time, so maybe it needs upgrading some dependency versions.

I need a few days to try to reproduce your issue on Windows machine.

About RT-DETR, no, I haven't tried it.

**Comment by jericko-ag** [comment]
Thanks @adalkiran!

**Comment by adalkiran** [comment]
Hi @jericko-ag , sorry for late update. I checked out the problem, and found that there are some problems with inference (Python) and mediabridge (Go) sides.

- Inference side: YOLOX's used version seems deprecated, because there are big changes in its dependencies. They've published newer version, but it can't be downloaded from PyPI, we need to download it from github.
Then, there's another issue with the latest version, one dependency version of YOLOX's new version must be downgraded (https://github.com/Megvii-BaseDetection/YOLOX/issues/1781#issuecomment-2243360660)
So, I need to do these fixes in Dockerfile

- Mediabridge side: After my latest commit (a long time ago), it seems browsers started to prioritize IPv6 addresses for local addresses. But the browser and the mediabridge can't match on a candidate pair.
I've solved this issue with a trick, but the end user must wait until all other candidates have failed. I've achieved to process frames successfully, but with these tricks, this is unacceptable to wait so long time to start.

I'll be working on this issue, I'm sure I'll find better method to fix.

Thanks for your patience!
